0000-catalog.md

```diff
@@ -0,0 +1,801 @@
+
+The CatalogProductSetStatus action will accept a parameter list of the
+catalogs the action should be performed on. Meaning Activate, Deactivate, or 
+Discontinue can be performed on a single, multiple, or all catalogs that an 
+organization has.
+
+## Permissions
+
+Creation of a Grid Catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field. An organization must have the GS1
+company prefix in the "gs1_company_prefixes" metadata field for its [Pike](https://grid.hyperledger.org/docs/grid/nightly/master/transaction_family_specifications/pike_transaction_family.html) organization.
+
+Deletion of a catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field. A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator. Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
+_Disclaimer: CatalogDelete & CatalogProductDelete are potentially hazardous
+operation and need to be done with care._
+
+To perform any of the **catalog actions** agents will need the following
+permissions:
+
+```
+CatalogCreate: Agent from org needs the "can_create_catalog" permission
+CatalogUpdate: Agent from org needs the "can_update_catalog" permission
+CatalogDelete: Agent from org needs the "can_delete_catalog" permission
+```
+
+To perform any of the **catalog_product actions** agents inherit the same
+permissions needed for a Grid Product:
+
+```
+CatalogProductCreate: Agent from org needs the "can_create_product" permission
+CatalogProductUpdate: Agent from org needs the "can_update_product" permission
+CatalogProductDelete: Agent from org needs the "can_delete_product" permission
+```
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+
+```
+CatalogProductSetStatus: Agent from org needs the "can_set_catalog_product_status" 
+permission
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+
+### Catalog Representation
+
+The primary object stored in state is `Catalog`, which consists of a
+`catalog_id` (hash of the catalog_name), an `owner` (org_id
+compatible w/Pike), a `name`, and a repeated field of `PropertyValues`. 
+The properties available are defined by the Grid Property Schema Transaction 
+Family and are restricted to the fields and rules of the GS1 Catalog schema 
+(which will be defined at a later time).  Transactions which are responsible 
+for setting catalog state values must ensure that those properties conform with 
+the requirements of the GS1 Catalog Schema (not yet defined).
+
+```
+message Catalog {
+    string catalog_id = 1;
+    string owner = 2;
+    string name = 3;
+    repeated PropertyValue properties = 4;
```

Comment:
So these properties are extended to any Product that is part of the Catalog. Is that correct?

Reply:
What do you mean?

The properties here are related to the catalog object itself, and not any products associated with the catalog. 

An example could be adding an "expiry_date" field to a catalog or an "issued_date" field.